src/nifreeze/utils/iterators.py

```diff
@@ -53,23 +54,24 @@ def linear_iterator(size: int = None, **kwargs) -> Iterator[int]:
     if size is None:
         raise TypeError("Cannot build iterator without size")
 
-    return range(size)
+    return iter(range(size))
```

Comment:
Like for the other iterators, let's return a generator:

```suggestion
    return (s for s in range(size))
```

Reply:
We should maybe spare some more thought to this: the functions being named `*_iterator` is misleading now, and the return type docstrings are misleading as well, as they all say `:obj:~typing.Iterator`. To avoid too many back and forths, and as I have not fully understood https://github.com/nipreps/nifreeze/pull/29#discussion_r1916292026, better to discuss this live over a call.